tests/test_run_service.py

```diff
@@ -0,0 +1,376 @@
+        file_handler.setLevel(logging.DEBUG)
+        file_formatter = logging.Formatter(
+            '%(asctime)s - %(message)s',
+            datefmt='%Y-%m-%d %H:%M:%S'
+        )
+        file_handler.setFormatter(file_formatter)
+        logger.addHandler(file_handler)
+    
+    return logger
+
+# Test Configuration
+TEST_CONFIG = {
+    "RPC_URL": os.getenv('RPC_URL', ''),
+    "BACKUP_WALLET": os.getenv('BACKUP_WALLET', '0x4e9a8fE0e0499c58a53d3C2A2dE25aaCF9b925A8'),
+    "TEST_PASSWORD": os.getenv('TEST_PASSWORD', ''),
+    "PRIVATE_KEY": os.getenv('PRIVATE_KEY', ''),
+    "STAKING_CHOICE": os.getenv('STAKING_CHOICE', '1')
+}
+
+# Expected prompts and their responses
+PROMPTS = {
+    r"eth_newFilter \[hidden input\]": TEST_CONFIG["RPC_URL"],
+    "input your password": TEST_CONFIG["TEST_PASSWORD"],
+    "confirm your password": TEST_CONFIG["TEST_PASSWORD"],
+    "Enter your choice": TEST_CONFIG["STAKING_CHOICE"],
+    "backup owner": TEST_CONFIG["BACKUP_WALLET"],
+    "Press enter to continue": "\n",
+    "press enter": "\n",
+    "Please make sure master EOA.*has at least.*xDAI": handle_xDAIfunding,
+    r"Enter local user account password \[hidden input\]": TEST_CONFIG["TEST_PASSWORD"]
+}
+
+class TestService:
+    @classmethod
+    def setup_class(cls):
+        """Setup for all tests"""
+        timestamp = datetime.now().strftime('%Y%m%d_%H%M%S')
+        cls.log_file = Path(f'test_run_service_{timestamp}.log')
+        cls.logger = setup_logging(cls.log_file)
+        
+        # Start the service
+        cls.start_service()
+        # Wait for service to fully start
+        time.sleep(5)
+        
+    @classmethod
+    def teardown_class(cls):
+        """Cleanup after all tests"""
+        try:
+                # First stop the service
+            cls.stop_service()
+            
+            # Wait for service to fully stop
+            time.sleep(10)
+            
+            # Then clean up the folder
+            cleanup_previous_run(cls.logger)
+        except Exception as e:
+            cls.logger.error(f"Error during cleanup: {str(e)}")
+        
+    @classmethod
+    def start_service(cls):
+        """Start the service and handle initial setup."""
+        try:
+            cls.logger.info("Starting run_service.py test")
+            
+            # Get current virtual environment path
+            venv_path = os.environ.get('VIRTUAL_ENV')
+            
+            # Temporarily deactivate virtual environment
+            if venv_path:
+                temp_env = os.environ.copy()
+                temp_env.pop('VIRTUAL_ENV', None)
+                temp_env.pop('POETRY_ACTIVE', None)
+                
+                # Update PATH to remove the virtual environment
+                paths = temp_env['PATH'].split(os.pathsep)
+                paths = [p for p in paths if not p.startswith(venv_path)]
+                temp_env['PATH'] = os.pathsep.join(paths)
+            else:
+                temp_env = os.environ
```

Comment:
We've kept the _setup_environment as a separate classmethod in TestService class because:
- It makes the code more maintainable and reusable
- Other test classes can inherit from TestService and reuse this setup
- It's cleaner to separate environment setup from test-specific setup